packages/core/src/indices/vectorStore/VectorStoreIndex.ts

```diff
@@ -50,15 +50,27 @@ export class VectorStoreIndex extends BaseIndex<IndexDict> {
       options.serviceContext ?? serviceContextFromDefaults({});
     const docStore = storageContext.docStore;
     const vectorStore = storageContext.vectorStore;
+    const indexStore = storageContext.indexStore;
+    
+    // Setup IndexStruct from storage
+    let indexStructs = await indexStore.getIndexStructs() as IndexDict[];
+    let indexStruct: IndexDict | null;
+    if (indexStructs.length == 1) {
+      indexStruct = indexStructs[0];
+    } else if (indexStructs.length > 1 && options.indexID) {
+      indexStruct = await indexStore.getIndexStruct(options.indexID) as IndexDict;
+    } else if (options.indexStruct) {
```

Comment:
OK so if I look at the Python code, load_index_from_storage is outside the default index initialization code path.

I think having it like this would work, although we probably want to make it clear this is what's going on.

For example, here if someone passes in an indexStruct in the initialization options it will be ignored if the store already has an indexStruct from persistence, which feels a bit not right.

Maybe we should at least move this else if option to the top, before we even check the store for existing index structs?